libs/@local/graph/postgres-store/src/store/postgres/seed_policies.rs

```diff
@@ -0,0 +1,333 @@
+use core::iter;
+
+use error_stack::{Report, ResultExt as _};
+use hash_graph_authorization::{
+    AuthorizationApi,
+    policies::{
+        Effect,
+        action::ActionName,
+        principal::PrincipalConstraint,
+        resource::{
+            EntityResourceConstraint, EntityResourceFilter, EntityTypeResourceConstraint,
+            EntityTypeResourceFilter, ResourceConstraint,
+        },
+        store::{
+            PolicyCreationParams, PolicyFilter, PolicyStore as _, PrincipalFilter,
+            error::EnsureSystemPoliciesError,
+        },
+    },
+};
+use type_system::{
+    ontology::{BaseUrl, VersionedUrl, id::OntologyTypeVersion},
+    principal::{
+        actor::{ActorId, ActorType},
+        role::{RoleId, TeamRole, WebRole},
+    },
+};
+
+use super::{AsClient, PostgresStore};
+
+macro_rules! base_url {
+    ($url:expr) => {
+        BaseUrl::new($url.to_owned()).expect("should be a valid base URL")
+    };
+}
+
+/// Creates a list of filters for entities with their type versions from 1 to `max_version`.
+// TODO: Allow entity filter for only the BaseURL
+//   see https://linear.app/hash/issue/H-4599/support-baseurl-and-version-filter-of-types-in-entity-resource
+fn create_version_filters(
+    base_url: BaseUrl,
+    max_version: u32,
+) -> impl Iterator<Item = EntityResourceFilter> {
+    let mut base_url = Some(base_url);
+
+    (1..=max_version).filter_map(move |version| {
+        Some(EntityResourceFilter::IsOfType {
+            entity_type: VersionedUrl {
+                base_url: if version < max_version {
+                    base_url.as_ref()?.clone()
```

Comment:
I think a `unwrap_or(|| unreachable!())` would be more intentional here than a `filter_map`. The value not being present would be a logic/programming error in this specific function and therefore you'd leak such an implementation error to the consumer. I mean the `None` case should never happen, no?

Reply:
Previously, I actually had `(1..)` instead of `(1..=max_version)` but clippy complained about an infinite loop 😂 
Fixed in 45c4f6c370fb2b803f81d32cdd9e8d5f1f7e49e9

Reply:
And c872515ffa556117e8ee92563c39ca853942e7ff